service/src/main/java/org/broadinstitute/listener/relay/http/RelayedHttpRequestProcessor.java

```diff
@@ -83,6 +97,38 @@ public Result writeNotAcceptedResponseOnCaller(RelayedHttpListenerContext contex
     return Result.FAILURE;
   }
 
+  public Result writePreflightResponse(RelayedHttpListenerContext context) {
+    if (context.getResponse() == null) {
+      logger.error("The context did not have a valid response");
+      return Result.FAILURE;
+    }
+
+    RelayedHttpListenerResponse listenerResponse = context.getResponse();
+    listenerResponse.setStatusCode(204);
+    listenerResponse
+        .getHeaders()
+        .put(ACCESS_CONTROL_ALLOW_METHODS, corsSupportProperties.preflightMethods());
+
+    if (!listenerResponse.getHeaders().containsKey(ACCESS_CONTROL_ALLOW_ORIGIN))
+      listenerResponse
+          .getHeaders()
+          .put(
+              ACCESS_CONTROL_ALLOW_ORIGIN,
+              context.getRequest().getHeaders().getOrDefault("Origin", "*"));
```

Comment:
As per the CORS spec, we should return a 400 status (Bad request), instead of  `*`, with `Reason: CORS header 'Origin' cannot be added to `when the Origin header missing`. Ref: [CORS header 'Origin'...](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSOriginHeaderNotAdded)

Reply:
that's probably true if we're trying to enforce CORS..but here, we're basically trying to allow the restriction, let's say user is using a browser that doesn't send the header, are we just gonna say "Terra doesn't work on your browser"?